tendermint/src/rpc/id.rs

```diff
@@ -3,9 +3,17 @@
 use getrandom::getrandom;
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq, Ord, PartialOrd)]
+#[serde(untagged)]
+pub enum IdType {
+    Num(i64),
+    Str(String),
+    None,
```

Comment:
Cool! That will make the jsonrpc id field spec compliant (see https://github.com/informalsystems/tendermint-rs/issues/88#issuecomment-599227891). I'd vouch to implement this in a separate PR if possible. Maybe testing it in a similar fashion as @romac did in #188 (see https://github.com/informalsystems/tendermint-rs/pull/188/files#diff-a1644bb72a1c83c54d82738ef71ce095R107).

Reply:
This is also a requirement for v0.33 compatibility, as it turns out. We agreed before that I'll implement this separately but then all of Shivani's JSON test files need to be updated for compatibility and I wasn't planning to do that in this PR. Tests are still moving around a lot and making all of them comply strictly with a broken implementation just to be able to push this PR through didn't make sense.

The change is in it's own commit, we can cherry-pick it into its own branch and dissect it more. But then this PR will be dependent on that one - OR an additional 10+ files need to be changed in this one to get away from that dependency.

Reply:
> This is also a requirement for v0.33 compatibility, as it turns out.

That this can deal with all 3 cases (number, string, null) now? OK.

I would still have been in favour of implementing this in a separate PR. Maybe even against this branch (not against master). Just because it is a clear small chunk of work that can be reviewed on it's own (now it's already in two commits). Next time :-)

Reply:
I can still break it out into its own thing. :)

Reply:
I'll let you decide. Looks like the id isn't used anywhere currently. At this stage maybe capturing what needs to be followed up with in an issue is less work then reverting, cherrypicking and opening a separate PR for it.

Reply:
Submitted PR #202 against master that makes this happen separately, as you asked. We'll have to make sure to resolve the CHANGES.md conflict here, but that's about it. (The rest were cherry-picked.)